lib/flipper/api/v1/actions/percentage_of_actors_gate.rb

```diff
@@ -0,0 +1,57 @@
+require 'flipper/api/action'
+require 'flipper/api/v1/decorators/feature'
+
+module Flipper
+  module Api
+    module V1
+      module Actions
+        class PercentageOfActorsGate < Api::Action
+          route %r{api/v1/features/[^/]*/percentage_of_actors/?\Z}
+
+          def post
+            ensure_valid_enable_params
+            feature = flipper[feature_name]
+            feature.enable_percentage_of_actors(percentage)
+            decorated_feature = Decorators::Feature.new(feature)
+            json_response(decorated_feature.as_json, 200)
+          end
+
+          def delete
+            ensure_valid_disable_params
+            feature = flipper[feature_name]
+            feature.disable_percentage_of_actors
+            decorated_feature = Decorators::Feature.new(feature)
+            json_response(decorated_feature.as_json, 200)
+          end
+
+          private
+
+          def ensure_valid_enable_params
+            json_error_response(:feature_not_found) unless feature_names.include?(feature_name)
+            json_error_response(:percentage_invalid) unless percentage >= 0 && percentage <= 100
+          end
+
+          def ensure_valid_disable_params
+            json_error_response(:feature_not_found) unless feature_names.include?(feature_name)
+          end
+
+          def feature_name
+            @feature_name ||= Rack::Utils.unescape(path_parts[-2])
+          end
+
+          def percentage
+            @percentage ||= begin
+                              Integer(params['percentage'])
+                            rescue ArgumentError, TypeError
```

Comment:
I like this much better great suggestion.  Here we rescue TypeError as well for nil cases Integer(nil)